docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/FieldAwareFactorizationMachine.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using Microsoft.ML.Data;
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class FFMBinaryClassification
+    {
+        public static void Example()
+        {
+            // Creating the ML.Net IHostEnvironment object, needed for the pipeline.
+            var mlContext = new MLContext();
+
+            // Download and featurize the dataset.
+            (var trainData, var testData) = SamplesUtils.DatasetUtils.LoadFeaturizedSentimentDataset(mlContext);
+
+            // ML.NET doesn't cache data set by default. Therefore, if one reads a data set from a file and accesses it many times, it can be slow due to
+            // expensive featurization and disk operations. When the considered data can fit into memory, a solution is to cache the data in memory. Caching is especially
+            // helpful when working with iterative algorithms which needs many data passes. Since SDCA is the case, we cache. Inserting a
+            // cache step in a pipeline is also possible, please see the construction of pipeline below.
+            trainData = mlContext.Data.Cache(trainData);
+
+            // Step 2: Pipeline
+            // Create the 'FieldAwareFactorizationMachine' binary classifier, setting the "Sentiment" column as the label of the dataset, and 
+            // the "Features" column as the features column.
+            var pipeline = new EstimatorChain<ITransformer>().AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.BinaryClassification.Trainers.
+                FieldAwareFactorizationMachine(labelColumnName: "Sentiment", featureColumnNames: new[] { "Features" }));
+
+            // Fit the model.
+            var model = pipeline.Fit(trainData);
+
+            // Let's get the model parameters from the model.
+            var modelParams = model.LastTransformer.Model;
+
+            // Let's inspect the model parameters.
+            var featureCount = modelParams.GetFeatureCount();
+            var fieldCount = modelParams.GetFieldCount();
+            var latentDim = modelParams.GetLatentDim();
+            var linearWeights = modelParams.GetLinearWeights();
```

Comment:
>var linearWeights = modelParams.GetLinearWeights(); [](start = 12, length = 51)

I think right now I can do following:
`Evaluate(model.Transform(Data)) -> AUC = X1`

```
var linearWeights = modelParams.GetLinearWeights();
linearWeights[0] =100;
linearWeights[1] =200;
```

`Evaluate(model.Transform(Data)) -> AUC = X2`

X1 != X2

Which is awful.

Can you check `MatrixFactoraztionPredictor` and how it handles arrays?

I also don't understand `GetFeatureCoun()` functions, why can't I just do `modelParams.FeatureCount`?